### PR TITLE
Merge 6.2.x into 7.0.x

### PR DESCRIPTION
**Pull request by jamesfredley**
@matrei @jdaugherty I am confident in the changes coming over from https://github.com/grails/grails-forge/pull/424 and their application to code that was not present on 6.2.x.  This is the vast majority of this PR.

Since 6.2.x and 7.0.x branches had diverged significantly, please focus on your recent commits and verify that I chose correctly when addressing the merge conflicts and decided correctly for 7.0.x.

```
# Conflicts:
#	.github/workflows/mavenCentral.yml
#	.github/workflows/release.yml
#	.github/workflows/snapshot.yml
#	gradle.properties
#	grails-cli/src/test/groovy/org/grails/forge/cli/CommandSpec.groovy
#	grails-forge-core/src/main/java/org/grails/forge/build/dependencies/Dependency.java
#	grails-forge-core/src/main/java/org/grails/forge/build/dependencies/Scope.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/assetPipeline/AssetPipeline.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/build/gradle/Gradle.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/build/gradle/templates/buildGradle.rocker.raw
#	grails-forge-core/src/main/java/org/grails/forge/feature/build/gradle/templates/buildSrcBuildGradle.rocker.raw
#	grails-forge-core/src/main/java/org/grails/forge/feature/build/gradle/templates/dependencies.rocker.raw
#	grails-forge-core/src/main/java/org/grails/forge/feature/build/gradle/templates/settingsGradle.rocker.raw
#	grails-forge-core/src/main/java/org/grails/forge/feature/micronaut/MicronautInjectGroovy.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/other/MicronautHttpClient.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/test/GebWithWebDriverBinaries.java
#	grails-forge-core/src/main/java/org/grails/forge/feature/view/Scaffolding.java
#	grails-forge-core/src/main/resources/pom.xml
#	grails-forge-core/src/test/groovy/org/grails/forge/build/gradle/GradleSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/asciidoctor/AsciidoctorSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/assetPipeline/AssetPipelineSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/database/HibernateGormSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/grails/GrailsGradlePluginSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/lang/GrailsApplicationSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/micronaut/MicronautHttpClientSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/micronaut/MicronautInjectGroovySpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/test/GebSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/view/GrailsGspSpec.groovy
#	grails-forge-core/src/test/groovy/org/grails/forge/feature/view/ScaffoldingSpec.groovy
#	test-core/src/test/groovy/org/grails/forge/features/scaffolding/ScaffoldingSpec.groovy
```

### CODE REVIEW

File: grails-forge-core/src/main/java/org/grails/forge/build/gradle/GradleDependency.java [hunk 1]
```diff
@@ -99,14 +101,15 @@ public int hashCode() {
 
     @NonNull
     public String toSnippet() {
-        String snippet = gradleConfiguration.getConfigurationName();
+        String snippet = gradleConfiguration.getConfigurationName() + " ";
         if (isPom()) {
-            String platformPrefix = " ";
-            snippet += platformPrefix + "platform";
+            snippet += "platform(";
+        } else if (gradleConfiguration == INTEGRATION_TEST_IMPLEMENTATION_TEST_FIXTURES) {
```
Comment (jamesfredley) [inline grails-forge-core/src/main/java/org/grails/forge/build/gradle/GradleDependency.java:107]: If we ever add another Gradle scope that requires () we will re-engineer this code at that time.

**Review by jdaugherty — COMMENTED** [review]
Spot checking this, it looks okay on the JDK version change.  @matrei  did most of the merges so I'd prefer he approve this one.

File: build.gradle [hunk 2]
```diff
@@ -27,4 +27,9 @@ if (!ossUser.isEmpty() && !ossPass.isEmpty()) {
             }
         }
     }
+}
+
+// buildSrc tests fail occasionally without running clean first
+build {
+    dependsOn clean
```
Comment (matrei) [inline build.gradle:34]: Where is this code coming from. I can't find it, neither in `6.2.x` or `7.0.x`.

File: grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java [hunk 3]
```diff
@@ -75,23 +75,26 @@ public void apply(GeneratorContext generatorContext) {
...
         generatorContext.addDependency(Dependency.builder()
                 .groupId("org.grails.plugins")
                 .artifactId("views-json")
-                .compile());
+                .implementation());
         generatorContext.addDependency(Dependency.builder()
                 .groupId("org.grails.plugins")
                 .artifactId("views-json-templates")
-                .compile());
+                .implementation());
```
Comment (matrei) [inline grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java:92]: I'm not sure the `views-json-templates` dependency should be included by default with the `views-json` feature.
I think the templates in `view-json-templates` might only be relevant when using MongoDB.
But that can be the topic for another PR.

**Review by matrei — APPROVED** [review]
LGTM :+1:

File: grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java (hunk 3 above)
Comment (jamesfredley) [reply grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java:92]: Sounds good.

File: build.gradle (hunk 2 above)
Comment (jamesfredley) [reply build.gradle:34]: It is technically new code that resolved an issue seen during the merge.  It can also occur on 6.2.x, so I will commit it over there and then update this PR before merging.

File: grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java (hunk 3 above)
Comment (jamesfredley) [reply grails-forge-core/src/main/java/org/grails/forge/feature/view/json/ViewJson.java:92]: https://github.com/grails/grails-forge/issues/443
